OpenRA.Mods.Common/Traits/Buildable.cs

```diff
@@ -38,10 +38,13 @@ public class BuildableInfo : TraitInfo<Buildable>
 		[Desc("Sequence of the actor that contains the icon.")]
 		public readonly string Icon = "icon";
 
-		[PaletteReference]
+		[PaletteReference("IsPlayerPalette")]
 		[Desc("Palette used for the production icon.")]
 		public readonly string IconPalette = "chrome";
 
+		[Desc("Custom palette is a player palette BaseName")]
+		public readonly bool IsPlayerPalette = false;
```

Comment:
The palette field is called `IconPalette` rather than `Palette`, so we probably want to call this `IconPaletteIsPlayerPalette` to avoid ambiguity/confusion...?

Reply:
Sure, why not.